### PR TITLE
[DataLoader] Fix util ImportError

### PR DESCRIPTION
**Pull request by ejguan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#52459 [DataLoader] Fix util ImportError**

Differential Revision: [D26523941](https://our.internmc.facebook.com/intern/diff/D26523941)

### CODE REVIEW

**Review by glaringlee — APPROVED** [review]
lgtm

File: torch/utils/data/datapipes/utils/decoder.py [hunk 1]
```diff
@@ -182,8 +199,16 @@ def torch_video(key, data):
...
         return None
 
-    # add `type: ignore` to avoid mypy's warning on import missing
-    import torchvision.io  # type: ignore
+    try:
+        import torchvision.io
+    except ImportError as e:
+        try:
+            import pip  # type: ignore
+            pip.main(['install', '--user', 'torchvision'])
```
Comment (VitalyFedyunin) [inline torch/utils/data/datapipes/utils/decoder.py:207]: Please DO NOT install things on users behalf, throw error with instruction how to do it

**Review by VitalyFedyunin — CHANGES_REQUESTED** [review]
Not not auto-install things, it is not secure, might screw-up users setup etc etc

File: torch/utils/data/datapipes/utils/decoder.py (hunk 1 above)
Comment (ejguan) [reply torch/utils/data/datapipes/utils/decoder.py:207]: It makes sense.

File: torch/utils/data/datapipes/utils/decoder.py [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+import os
 import pickle
 import re
-import os
+import tempfile
 
 import json
-import tempfile
-import io
+import numpy as np
```
Comment (ejguan) [inline torch/utils/data/datapipes/utils/decoder.py:11]: `numpy` is one of the requirements for pytorch, so it should be fine for direct import

**Review by VitalyFedyunin — APPROVED** [review]

**Pull request closed without merging**